packages/block-library/src/list/list-type-picker.js

```diff
@@ -0,0 +1,81 @@
+/**
+ * External dependencies
+ */
+import { map } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { Dashicon, BaseControl, Button, Dropdown, NavigableMenu } from '@wordpress/components';
+
+function ListTypePicker( {
+	listTypes = [],
+	onChange,
+	value,
+} ) {
+	if ( ! listTypes.length ) {
+		return null;
+	}
+
+	const currentType = listTypes.find( ( listType ) => listType.type === value );
+	const currentListTypeName = ( currentType && currentType.name );
+
+	return (
+		<BaseControl label={ __( 'List Type' ) }>
```

Comment:
@kjellr or @jasmussen I see that this follows the same pattern FontSizePicker introduced. There is also SelectControl:
https://github.com/WordPress/gutenberg/tree/master/packages/components/src/select-control

What’s the difference between those two? Should we extract duplicated logic and create another UI component?

Reply:
> What’s the difference between those two? 

Yeah, this is a bit tricky actually. I'm not completely sure. The FontSizePicker control has instant visual effect, which is desirable here. I would suggest that's why it looks more like a button than a dropdown. Whereas the SelectControl (traditionally, dropdown), does not always have instant visual effect. Could they be the same? Not sure, but it does seem to make sense to either merge them or distinguish them more. Perhaps the FontSizePicker component should have more in common, both visually and technically, with the formatting dropdowns in the block toolbar, like the inline text formatting:

<img width="307" alt="Screenshot 2019-04-25 at 11 01 14" src="https://user-images.githubusercontent.com/1204802/56723551-74b80000-6749-11e9-91fb-5295f90fafa8.png">

Though I would suggest it's not critical that happen in this particular PR.

Reply:
It seems like in this case `SelectControl` would be perfectly fine. I don't see any custom icons used next to options to select.

Reply:
The select control has been added